### PR TITLE
Retain reqdata for use in #dispatch_host notification

### PR DESCRIPTION
**Pull request by Dirklectisch**
This is a small fix for missing request data when observing #dispatch_host.

### CODE REVIEW

File: src/support/z_sites_dispatcher.erl [hunk 1]
```diff
@@ -587,7 +587,8 @@ find_dispatch_fallback() ->
 first_site_match([], _DispHost, _ReqData) ->
     no_host_match;
 first_site_match([Site|Sites], DispHost, ReqData) ->
-    case catch z_notifier:first(DispHost, z_context:new(Site)) of
+    NewSiteContext = z_context:new(Site),
+    case catch z_notifier:first(DispHost, NewSiteContext#context{wm_reqdata=ReqData}) of
```
Comment (mworrell) [inline src/support/z_sites_dispatcher.erl]: The `z_context:new/1` call should be inside the catch expression.
You should use `z_context:set_reqdata(ReqData, z_context:new(Site)))` here.

**Comment by mworrell** [comment]
I made a similar change in the master https://github.com/zotonic/zotonic/commit/588a9035d037e092f5c7ef82c94147b1b0e0ebd0

Can you fix your merge request?

**Comment by Dirklectisch** [comment]
Updated with your suggestion.

**Comment by mworrell** [comment]
Thanks!

**Pull request merged**